modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/MessageDeliveryTask.java

```diff
@@ -73,7 +73,7 @@ public TaskHint call() throws Exception {
 
             // Get a slot from coordinator.
             Slot currentSlot = requestSlot(storageQueueName);
-
+            currentSlot.setStorageQueue(storageQueue);
```

Comment:
Shouldn't the coordinator set this and return? If we have to set the storage queue separately like this it implies that something is not right in the slot coordinator when returning a slot since we pass the storage queue when requesting a slot.